modules/openapi-generator/src/main/resources/typescript-fetch/modelEnum.mustache

```diff
@@ -1,10 +1,16 @@
 {{>modelEnumInterfaces}}
 
-export function {{classname}}FromJSON(json: any): {{classname}} {
+export function {{classname}}FromJSON(json: any): {{classname}} | null {
```

Comment:
this is gonna be a breaking change. if an enum property is marked as required / non-nullable in the spec, all consumers expect that it's defined.
after this change some apps will likely stop compiling.

why do yo need it anyway? you can keep the original code here  - return the enum value as is, without any processing

Reply:
Typescript gave a me a warning about the implicit return null. I believe it can be removed

Reply:
that's because this PR changed the code inside the function, narrowing down `json` type to `null` before returning it

Reply:
Playing around with this more... it previously returned a value that was no valid. so if you passed `7` and your enum was `[0, 1, 2]` you would get back 7 which is not the right type.

Reply:
well, we trust the api response matches the spec. otherwise the code is gonna break anyway, since we don't do full shape verification

Reply:
The challenge is older clients being used with newer backends which just weren't aware of the existence of such values

Reply:
I would say, `null` is as unexpected as a bogus enum value (if the property is not marked as nullable).
Again, i honestly believe it's way out of scope of the generated api client.